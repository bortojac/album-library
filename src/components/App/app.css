.app {
  height: 100%;
  width: 100%;
}

.app__main {
  display: flex;
  flex-flow: column;
  height: calc(100% - 4rem);
  align-items: center;
}