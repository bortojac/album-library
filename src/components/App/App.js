import React, { Component } from 'react';
import Header from '../Header/Header';
import Search from '../Search';
import Library from '../Library/Library';
import './app.css';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      searchActive: false,
    };
    this.handleToggleSearch = this.handleToggleSearch.bind(this);
  }

  componentDidMount() {
    this.props.getAccessToken();
  }

  handleToggleSearch() {
    this.setState({ searchActive: !this.state.searchActive });
  }

  render() {
    return (
      <div className="app">
        <Header toggleSearch={this.handleToggleSearch}/>
        <main className="app__main">
          <Search active={this.state.searchActive} toggleSearch={this.handleToggleSearch}/>
          <Library />
        </main>
      </div>
    );
  }
}

export default App;
