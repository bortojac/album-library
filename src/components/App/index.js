import { connect } from 'react-redux';
import App from './App';
import { getAccessToken } from '../../store/actions';

const mapStateToProps = state => ({ })

const mapDispatchToProps = dispatch => {
  return {
    getAccessToken: () => {
      dispatch(getAccessToken())
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
