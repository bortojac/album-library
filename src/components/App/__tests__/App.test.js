import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import App from '../App';

//jest.mock('../../Search', () => 'Search'); // gives unknown prop warnings

jest.mock('../../Search'); // use manual mock in __mocks__ folder of Search

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App getAccessToken={()=>''}/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const tree = renderer
    .create(<App getAccessToken={()=> ''}  />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
