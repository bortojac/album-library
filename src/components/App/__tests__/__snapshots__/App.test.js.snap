// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="app"
>
  <div
    className="header"
  >
    <h1
      className="header__h1"
    >
      Album Library
    </h1>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-plus-circle fa-w-16 header__icon"
      data-icon="plus-circle"
      data-prefix="fas"
      onClick={[Function]}
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
  <main
    className="app__main"
  >
    <div
      className="library"
    >
      <h1
        className="library__h1"
      >
        The Top Shelf
      </h1>
      <button
        className="library__button"
        onClick={[Function]}
        type="button"
      >
        Load more
      </button>
    </div>
  </main>
</div>
`;
